tests/test_sketchcomparison.py

```diff
@@ -270,6 +270,65 @@ def test_FracMinHashComparison_fail_threshold(track_abundance):
     assert not cmp.pass_threshold # threshold is 10; this should fail
 
 
+def test_FracMinHashComparison_potential_false_negative(track_abundance):
+    a = MinHash(0, 21, scaled=10000, track_abundance=track_abundance)
+    b = MinHash(0, 21, scaled=10000, track_abundance=track_abundance)
+    c = MinHash(0, 21, scaled=10000, track_abundance=track_abundance)
+    # make sketch overlap quite small
+    a_values = {1:5}
+    b_values = {1:3, 3:2, 5:1, 6:1, 8:1, 10:1 }
+    c_values = {2:1}
+
+    if track_abundance:
+        a.set_abundances(a_values)
+        b.set_abundances(b_values)
+        c.set_abundances(c_values)
+    else:
+        a.add_many(a_values.keys())
+        b.add_many(b_values.keys())
+        c.add_many(c_values.keys())
+
+    # build FracMinHashComparison
+    cmp = FracMinHashComparison(a, b)
+    # check ani, potential false negative
+    cmp.estimate_jaccard_ani()
+    assert cmp.jaccard_ani == a.jaccard_ani(b).ani == b.jaccard_ani(a).ani
+    assert cmp.potential_false_negative == a.jaccard_ani(b).p_exceeds_threshold == b.jaccard_ani(a).p_exceeds_threshold
+    assert cmp.potential_false_negative == True
+    assert cmp.jaccard_ani_untrustworthy == a.jaccard_ani(b).je_exceeds_threshold == b.jaccard_ani(a).je_exceeds_threshold
+
+    cmp.estimate_mh1_containment_ani()
+    a_cont_ani_manual = a.containment_ani(b)
+    assert cmp.mh1_containment_ani == a_cont_ani_manual.ani
+    # potential false negative is false here. But since we got it above, don't want to set to False
+    assert cmp.potential_false_negative != a_cont_ani_manual.p_exceeds_threshold
+    assert cmp.potential_false_negative == True
+
+    cmp.estimate_mh2_containment_ani()
+    b_cont_ani_manual = b.containment_ani(a)
+    assert cmp.mh2_containment_ani == b_cont_ani_manual.ani
+    assert cmp.potential_false_negative == b_cont_ani_manual.p_exceeds_threshold
+    assert cmp.potential_false_negative == True
+
+    cmp.estimate_max_containment_ani()
+    mc_ani_manual = a.max_containment_ani(b)
+    assert cmp.max_containment_ani == max(a.containment_ani(b).ani, b.containment_ani(a).ani) == mc_ani_manual.ani
+    # potential false negative is false here. But since we got it above, don't want to set to False
+    assert cmp.potential_false_negative != mc_ani_manual.p_exceeds_threshold
+    assert cmp.avg_containment_ani == np.mean([a.containment_ani(b).ani, b.containment_ani(a).ani])
+    assert cmp.potential_false_negative == True
+
+    # comparison in opposite direction, so we can test potential_false_neg in other direction
+    cmp2 = FracMinHashComparison(b, a)
+    cmp2.estimate_mh1_containment_ani()
+    assert cmp2.potential_false_negative == True
+
+    # test max cont potenital_false_neg (via no overlap at all)
```

Comment:
```suggestion
    # test max cont potential_false_neg (via no overlap at all)
```

Reply:
> Looks good!
> 
> A question that I didn't see clearly addressed anywhere (but I might have missed!) is what happens when you try to calculate ANI for two very small scaled sketches? Does it just end up being 0, or None, or ?

At the moment, we just report the ANI, except for extremely tiny test data where we _can't_ actually estimate ANI. 

For jaccard --> ANI, we estimate the error on the jaccard estimate itself, and raise a warning when the error may be too high (but still currently report ANI). I have an item in the `SearchResult` class that keeps track of whether the jaccard estimate error is too high -- I think we should at least consider doing that, but would also be open to zeroing out the ANI estimate.

From on https://github.com/sourmash-bio/sourmash/issues/1798#issuecomment-1015762466:
> For example, using a scale factor of s=1/1000, if you want to be at least 95% sure that the FracMinHash cardinality is off by less than 5% relative error, you want to be sure that your set has more than ~4.4x10^6 elements.
> Interestingly, this approach will always be less space efficient than HLL. HLL experiences a relative error of something like 1.04/Sqrt(m) for a sketch of size m. For FracMinHash, you end up with a relative error of (at absolute minimum, often much worse) 2.07944/Sqrt(m) for a sketch of size m.

I was hoping we might be able to use HLL to avoid issues with small sketches, but I suppose instead we could use this to estimate an error based on the sketch size, and raise a warning when the error/ zero out the ANI when the error is too high?

Reply:
maybe just dump this in an issue and we can revisit later? I'm not sure there is a simple resolution here :)

Reply:
see #2003

Reply:
the only thing I'd consider doing here is adding the `jaccard_untrustworthy` or `potential_false_negative` to search output, but I am also open to doing that as part of #1555

Reply:
see #2004 :)